adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,43 @@
 #include "impl.h"
 #include <vector>
+#include<iostream>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    std::stack<char> br_stack;
+
+    for(const char &ch : source)
+    {
+
+        for(const auto &bracket : brackets )
+        {
+            if(br_stack.empty() && ch==bracket.second )
+            {
+                return false;
+            }
+
+            if(ch==bracket.first)
+            {
+                br_stack.push(ch);
+                break;
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es77-minimize-the-use-of-break-and-continue-in-loops